### PR TITLE
Adding CosmosSerializationOptions and partition key tests for query

### PR DESCRIPTION
**Pull request by j82w**
Adding tests to verify that the CosmosSerializationOptions are being applied and that query with a partition key works correctly.

Closes: #277

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs [hunk 1]
```diff
@@ -433,6 +432,37 @@ public async Task ItemMultiplePartitionQueryStream()
             }
         }
 
+        /// <summary>
+        /// Validate multiple partition query
+        /// </summary>
+        [TestMethod]
+        public async Task ItemSinglePartitionQueryStream()
+        {
+            IList<ToDoActivity> deleteList = await this.CreateRandomItems(101, randomPartitionKey: true);
```
Comment (bchong95) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: named parameters for both params please.

Comment (bchong95) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: If we allow partition key range id, can we also create a test for that?

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs [hunk 2]
```diff
@@ -433,6 +432,37 @@ public async Task ItemMultiplePartitionQueryStream()
...
+            CosmosSqlQueryDefinition sql = new CosmosSqlQueryDefinition("SELECT * FROM toDoActivity t");
+
+            ToDoActivity find = deleteList.First();
+            double totalRequstCharge = 0;
+            CosmosFeedIterator setIterator =
+                this.Container.Items.CreateItemQueryAsStream(sql, maxConcurrency: 1, partitionKey: find.status);
+            while (setIterator.HasMoreResults)
+            {
+                CosmosResponseMessage iter = await setIterator.FetchNextSetAsync();
```
Comment (bchong95) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: This only tests that the items found have the correct partition key right? We should also assert that found all the items to find.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs [hunk 3]
```diff
@@ -433,6 +432,37 @@ public async Task ItemMultiplePartitionQueryStream()
...
+        /// <summary>
+        /// Validate multiple partition query
+        /// </summary>
+        [TestMethod]
+        public async Task ItemSinglePartitionQueryStream()
+        {
+            IList<ToDoActivity> deleteList = await this.CreateRandomItems(101, randomPartitionKey: true);
+            CosmosSqlQueryDefinition sql = new CosmosSqlQueryDefinition("SELECT * FROM toDoActivity t");
+
+            ToDoActivity find = deleteList.First();
```
Comment (bchong95) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: Does this guarantee that we get can find multiple items within a logical partition?

**Comment by kirankumarkolli** [comment]
Please refresh PR description with more appropriate details. 
Also please ensure an issue (feature/bug) exists tracking it.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs (hunk 1 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: Partition Key range id is not supported in v3

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs (hunk 3 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: Updated the test to make multiple items with the same PK value.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs (hunk 2 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosItemTests.cs]: Done, in latest iteration.

**Pull request merged**